RecoPixelVertexing/PixelTriplets/plugins/CAHitNtupletGeneratorOnGPU.cc

```diff
@@ -141,12 +143,14 @@ void CAHitNtupletGeneratorOnGPU::fillDescriptions(edm::ParameterSetDescription&
   desc.add<bool>("fillStatistics", false);
   desc.add<unsigned int>("minHitsPerNtuplet", 4);
   desc.add<unsigned int>("maxNumberOfDoublets", CAConstants::maxNumberOfDoublets());
+  desc.add<unsigned int>("minHitsForSharingCut", 3)->setComment("Maximum number of hits in a tuple to clean also if the shared hit is on bpx1");
   desc.add<bool>("includeJumpingForwardDoublets", false);
   desc.add<bool>("fit5as4", true);
   desc.add<bool>("doClusterCut", true);
   desc.add<bool>("doZ0Cut", true);
   desc.add<bool>("doPtCut", true);
   desc.add<bool>("useRiemannFit", false)->setComment("true for Riemann, false for BrokenLine");
+  desc.add<bool>("doSharedHitCut", true)->setComment("Sharing hit nTuples cleaning");
```

Comment:
strictly speaking, to be consistent with previous behavior and with minHitsPerNtuplet default == 4, this should be false.

Still I do not object to have it "true" from now on...  (and loose some throughput for quadruplets)

Reply:
minor

Reply:
Right, indeed. I'd set `minHitsPerNtuplet == 99` to default so that if one wants to run triplets as they were should just turn on `doSharedHitCut`. Anyway I'm producing some validations and I'll attach them below to have a clear comparison and choose a *default* setup.

Reply:
I am a bit nervous that to switch from Quads to triplets we have now to change 4 configurables....

Reply:
Thinking out loud: once we choose the standard triplet configuration would it make sense to have a single flag that *overwrite* what needs to be overwritten. Is it heresy?

Reply:
I much prefer a single flag for production use

Reply:
One can add a LogWarning saying that defaults (and even user-settings?) are overwritten by the flag